### ISSUE TITLE
Response builder fluent API 

### ISSUE THREAD

**Issue by pmcelhaney** [pmcelhaney/counterfact#133]
Add a parameter to the verb functions called `response`. The API looks like this.

```js
response[statusCode]
  .header("a-string-header", "value")
  .header("a-number-header", 100)
  .match("content/type", "body")
  .match("another/type", "body")
  .json({ isJson: true }) // shortcut for .match("application/json", ...)
  .text("text") // shortcut for .match("text/plain", ...)
  .html("<p>html</p>"); // shortcut for .match("text/html", ...)
```

The return value is a response object.

```js
const response = {
   statusCode: statusCode,
   headers: { ... },
   contentType: {
        "text/plain": "text body",
         "application/json": {json: "body"}
    }
}
```

If someone else picks this up I'll add more detail. But I'll probably get to it myself today.

**Comment by pmcelhaney** [comment]
Depends on #134 

This one could be done first, but we'd have to loop back and add support for multiple content types.

**Comment by pmcelhaney** [comment]
`response[404]`, `response["404_NOT_FOUND"]` and `response["404 No pet found"]` should be equivalent because it will be useful when we start dealing with TypeScript and intellisense (#136). I'm not creating a separate issue because it only requires a one line change, adding `parseInt()` in the right place.
